test/index.test.ts

```diff
@@ -1,31 +1,50 @@
 import { strictEqual, deepStrictEqual } from "assert";
 import { describe, test } from "mocha";
-import { getAccounts, getConnection } from "@tableland/local";
+import { getAccounts, getDatabase, getValidator } from "@tableland/local";
 
 describe("index", function () {
   // Note that we're using the second account here
   const [, signer] = getAccounts();
-  const sdk = getConnection(signer);
+  const db = getDatabase(signer);
+  const validator = getValidator();
 
   test("create", async function () {
-    const { name } = await sdk.create("counter integer", { prefix: "table" });
-    strictEqual(name, "table_31337_2");
+    const { meta } = await db
+      .prepare("CREATE TABLE my_table (counter integer);")
+      .all();
+    strictEqual(meta.txn?.name, "my_table_31337_2");
   });
 
   test("insert", async function () {
-    const { hash } = await sdk.write("insert into table_31337_2 values (1);");
-    const txnReceipt = await sdk.receipt(hash);
+    const { meta } = await db
+      .prepare("insert into my_table_31337_2 values (1);")
+      .all();
+
+    strictEqual(typeof meta.txn?.transactionHash, "string");
+    strictEqual(typeof meta.txn?.chainId, "number");
```

Comment:
Because the `type` returned by `await db.prepare(...).all()` is unknown the developer should be checking that the javascript object returned matches their expectations.  However in the case of writing tests using a strict equal test as above doesn't satisfy the typescript transpiler.

Reply:
We could potentially do
```
if (typeof meta.txn?.transactionHash !== "string") throw new Error("test fails but in a weird way!");
if (typeof meta.txn?.chainId !== "number") throw new Error("test fails but in a weird way!");
```
This satisfies typescript, but might be a bit strange since the errors will never be thrown and tests would normally use an assertion for this kind of thing.

Reply:
The return type is `unknown` because we aren't typing it. But the prepare method is generic, so if you specify a type there, then it should be what we expect it to be. If this is not the case, then something is up? Or am I not understanding the question right?

Reply:
I think I'm just not understanding how to use typescript correctly in this case.
To put my issue another way:
`meta.txn?.transactionHash` has a type of 'string | undefined' [def](https://github.com/tablelandnetwork/js-tableland/blob/main/src/validator/client/validator.ts#L79)
and the `validator.receiptByTransactionHash` method expects an argument object with a transactionHash key that is `string`. I tried specifying the generic type for the prepare method, but I couldn't get that to work.
The two ways I've been able to coerce the type `string | undefined` to `string` is either by using a `typeof` conditional, or an `as` type assertion.  (These both work, just wondering if there's a better way?)

Reply:
Ah ok I see. Its type is actually derived here: https://github.com/tablelandnetwork/js-tableland/blob/main/src/validator/receipt.ts#L26. And `transactionHash` is actually a string. The reason you are getting `string | undefined` is because you are using the conditional chaining syntax: `meta.txn?.transactionHash` (note the question mark). The way to deal with this in typescript is to assert that `meta.txn` is not undefined. And then everything will show up as their correct type.

There is no way for us know know for sure ahead of time if `meta.txn` is defined. But you can do:

```typescript

if (meta.txn == null) {
    // throw or something
}

// then things will work here

```

Reply:
I think that makes sense.  Just to make sure I understand, I'll push a new commit with one test using `as` and the other test using `throw`

Reply:
You can also use assets, and typescript should be smart enough to deal with that correct. So you can so something like (or similar) and the type checking should know that any check _after_ this is going to be safe.

```
strictNotEquals(meta.txn, null);
```